antsibull/write_docs.py

```diff
@@ -163,6 +163,33 @@ async def write_collection_list(collections: t.Iterable[str], template: Template
         await f.write(index_contents)
 
 
+async def write_plugin_type_index(plugin_type: str,
+                                  collection_info: CollectionInfoT,
+                                  template: Template,
+                                  dest_filename: str) -> None:
+    """
+    Write an index page for each plugin type.
+
+    :arg plugin_type: The plugin type to write the index for.
+    :arg collection_info: Mapping of collection_name to Mapping of plugin_type to Mapping of
+        collection_name to short_description.
+    :arg template: A template to render the plugin index.
+    :arg dest_filename: The destination filename.
+    """
+    per_collection_data: t.Dict[str, t.Mapping[str, str]] = {
+        collection_name: plugin_maps[plugin_type]
+        for collection_name, plugin_maps in collection_info.items()
+        if plugin_type in plugin_maps
+    }
```

Comment:
This one block from line 179 should be done in the caller since it is transforming the data rather than formatting of output and writing of that output.